### ISSUE TITLE
D/TC:? 0 tee_ta_close_session:518 Destroy session D/TC:? 0 destroy_context:298 Destroy TA ctx (0xe178ac0)

### ISSUE THREAD

**Issue by zzyupupup** [OP-TEE/optee_os#5929]
Hi
		I built `OPTEE 3.8.0` on `QEMUV8`, and when I ran `optee_ example_ hello_ World` runs correctly in normal world

~~~
Invoking TA to increment 42
TA incremented value to 43
~~~

but the following error occurs in secure world:

~~~
D/TC:? 0 system_open_ta_binary:253 res=0xffff0008
D/TC:? 0 system_open_ta_binary:250 Lookup user TA ELF 8aaaf200-2450-11e4-abe2-0002a5d5c51b (REE)
D/TC:? 0 system_open_ta_binary:253 res=0x0
D/LD:  ldelf:169 ELF (8aaaf200-2450-11e4-abe2-0002a5d5c51b) at 0x40075000
D/TC:? 0 tee_ta_close_session:499 csess 0xe178320 id 1
D/TC:? 0 tee_ta_close_session:518 Destroy session
D/TC:? 0 tee_ta_close_session:499 csess 0xe178b20 id 1
D/TC:? 0 tee_ta_close_session:518 Destroy session
D/TC:? 0 destroy_context:298 Destroy TA ctx (0xe178ac0)
~~~

Similarly, when I run `optee_ example_ random` runs correctly in normal world:

~~~
Invoking TA to generate random UUID... 
TA generated UUID value = 0xeed785d6171a7bf1583f695dd5b54170
~~~

but the following error occurs in secure world:

~~~
D/TC:? 0 tee_ta_init_pseudo_ta_session:280 Lookup pseudo TA b6c53aba-9669-4668-a7f2-205629d00f86
D/TC:? 0 load_ldelf:704 ldelf load address 0x40006000
D/LD:  ldelf:134 Loading TA b6c53aba-9669-4668-a7f2-205629d00f86
D/TC:? 0 tee_ta_init_session_with_context:573 Re-open TA 3a2f8978-5dc0-11e8-9c2d-fa7ae01bbebc
D/TC:? 0 system_open_ta_binary:250 Lookup user TA ELF b6c53aba-9669-4668-a7f2-205629d00f86 (Secure Storage TA)
D/TC:? 0 system_open_ta_binary:253 res=0xffff0008
D/TC:? 0 system_open_ta_binary:250 Lookup user TA ELF b6c53aba-9669-4668-a7f2-205629d00f86 (REE)
D/TC:? 0 system_open_ta_binary:253 res=0x0
D/LD:  ldelf:169 ELF (b6c53aba-9669-4668-a7f2-205629d00f86) at 0x40016000
D/TC:? 0 tee_ta_close_session:499 csess 0xe178320 id 1
D/TC:? 0 tee_ta_close_session:518 Destroy session
D/TC:? 0 tee_ta_close_session:499 csess 0xe178b20 id 1
D/TC:? 0 tee_ta_close_session:518 Destroy session
D/TC:? 0 destroy_context:298 Destroy TA ctx (0xe178ac0)
~~~

When I run someone else's program, I should have output in the safe world as follow：

~~~
# I/TA:  loss = 1.62141, avg loss = 1.62540 from the TA
# I/TA:  loss = 1.58659, avg loss = 1.61783 from the TA
# I/TA:  loss = 1.57328, avg loss = 1.59886 from the TA
# I/TA:  loss = 1.52641, avg loss = 1.57889 from the TA
~~~

But I still got the following error：

~~~
D/TC:? 0 system_open_ta_binary:253 res=0xffff0008
D/TC:? 0 system_open_ta_binary:250 Lookup user TA ELF 7fc5c039-0542-4ee1-80af-b4eab2f1998d (REE)
D/TC:? 0 system_open_ta_binary:253 res=0x0
D/LD:  ldelf:169 ELF (7fc5c039-0542-4ee1-80af-b4eab2f1998d) at 0x40015000
D/TC:? 0 tee_ta_close_session:499 csess 0xe178320 id 1
D/TC:? 0 tee_ta_close_session:518 Destroy session
~~~

How can I modify it？

**Comment by jforissier** [comment]
Hello @zzyupupup, I do not see any error in the output you provide. The lines with `res=0xffff0008` mean the TA is not found as a "Secure Storage TA" but immediately after it is found in the REE filesystem.

**Comment by zzyupupup** [comment]
> 

~~~
root@Vexpress:/ optee_hello_world
Invoking TA to increment 42
TA incremented value to 43
~~~

The output of my running case in `OPTEE 2.6.0` in the secure world is as follows:

~~~
DEBUG:   USER-TA:TA_OpenSessionEntryPoint:70: has been called
INFO:    USER-TA: Hello World!
DEBUG:   USER-TA:inc_value:107: has been called
INFO:    USER-TA: Got value: 42 from NW
INFO:    USER-TA: Increase value to: 43
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:378: tee_ta_close_session(0xe077ed0)
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:397: Destroy session
INFO:    USER-TA: Goodbye!
FLOW:    USER-TA: tee_user_mem_free:442: Free: link:[0x117058], buf:[0x117068:16]
DEBUG:   USER-TA:TA_DestroyEntryPoint:52: has been called
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:423: Destroy TA ctx
~~~

`IMSG("Hello World!")` will be output，Then`tee_ta_close_session`,However, I did not output `Hello world` in `OPTEE 3.8`, and directly`tee_ta_close_session:518 Destroy session`

**Comment by zzyupupup** [comment]
> Hello @zzyupupup, I do not see any error in the output you provide. The lines with `res=0xffff0008` mean the TA is not found as a "Secure Storage TA" but immediately after it is found in the REE filesystem.

Thanks@jenswi-linaro
~~~
root@Vexpress:/ optee_hello_world
Invoking TA to increment 42
TA incremented value to 43
~~~

The output of my running case in `OPTEE 2.6.0` in the secure world is as follows:

~~~
DEBUG:   USER-TA:TA_OpenSessionEntryPoint:70: has been called
INFO:    USER-TA: Hello World!
DEBUG:   USER-TA:inc_value:107: has been called
INFO:    USER-TA: Got value: 42 from NW
INFO:    USER-TA: Increase value to: 43
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:378: tee_ta_close_session(0xe077ed0)
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:397: Destroy session
INFO:    USER-TA: Goodbye!
FLOW:    USER-TA: tee_user_mem_free:442: Free: link:[0x117058], buf:[0x117068:16]
DEBUG:   USER-TA:TA_DestroyEntryPoint:52: has been called
DEBUG:   [0x0] TEE-CORE:tee_ta_close_session:423: Destroy TA ctx
~~~

`IMSG("Hello World!")` will be output，Then`tee_ta_close_session`,However, I did not output information in `OPTEE 3.8`, and directly`tee_ta_close_session:518 Destroy session`
